headerfs/file.go

```diff
@@ -1,18 +1,211 @@
+//go:build !windows && !js && !wasm
+
 package headerfs
 
 import (
 	"bytes"
 	"fmt"
 	"os"
+	"path/filepath"
+	"sync"
 
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/walletdb"
 )
 
-// ErrHeaderNotFound is returned when a target header on disk (flat file) can't
-// be found.
-type ErrHeaderNotFound struct {
-	error
+// headerBufPool is a pool of bytes.Buffer that will be re-used by the various
```

Comment:
I think this would be easier to review if one commit moved the contents into this new file, then subsequent commits start to add the build tag and additional refactoring.